### PR TITLE
feat: add more granular fetch options for timeseries

### PR DESCRIPTION
**Pull request by ogzhanolguncu**
## What does this PR do?

- This PR adds more granular display for charts
- Turns logs table into generic table
- Fixes bugs caused by ui/button change

*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [Contributing Guide](./CONTRIBUTING.md)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand areas
- [ ] Ran `pnpm build`
- [ ] Ran `pnpm fmt`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `NoBreadcrumb` component as a placeholder.
	- Added a `LogsChart` component for visualizing log data.
	- Created a `LogsTable` component for displaying logs in a virtualized format.
	- Implemented a custom hook, `useFetchTimeseries`, for fetching timeseries log data.
	- Enhanced `LogsPage` to handle both logs and timeseries data.
  
- **Bug Fixes**
	- Improved error handling during data fetching processes.

- **Documentation**
	- Updated import paths for various components and types for better organization.

- **Chores**
	- Removed deprecated components and types to streamline the codebase.

### CODE REVIEW

File: apps/dashboard/app/(app)/logs/components/charts/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import { type TimeseriesGranularity, getTimeseriesGranularity } from "../../utils";
+
+const roundToSecond = (timestamp: number) => Math.floor(timestamp / 1000) * 1000;
+
+const formatTimestamp = (value: string | number, granularity: TimeseriesGranularity) => {
+  const date = new Date(value);
+  const offset = new Date().getTimezoneOffset() * -1;
+  const localDate = addMinutes(date, offset);
+
+  switch (granularity) {
```
Comment (chronark) [inline apps/dashboard/app/(app)/logs/components/charts/hooks.ts:15]: is the granularity passed straight from query params?
I'd prefer if we do `"hour" | "day" | ...`, it's consistent with other pages and our API

File: apps/dashboard/app/(app)/logs/components/charts/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      return format(localDate, "MMM d");
+    default:
+      return format(localDate, "Pp");
+  }
+};
+
+export const useFetchTimeseries = (initialTimeseries: LogsTimeseriesDataPoint[]) => {
+  const { searchParams } = useLogSearchParams();
+
+  const filters = useMemo(
```
Comment (chronark) [inline apps/dashboard/app/(app)/logs/components/charts/hooks.ts:30]: what does the useMemo do here?
isn't it the same as just accessing the values directly?

File: apps/dashboard/app/(app)/logs/components/charts/index.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+import type { LogsTimeseriesDataPoint } from "@unkey/clickhouse/src/logs";
+import { addMinutes, format } from "date-fns";
+import { useEffect, useState } from "react";
+import { Bar, BarChart, CartesianGrid, ResponsiveContainer, XAxis, YAxis } from "recharts";
+import { useFetchTimeseries } from "./hooks";
+
+const chartConfig = {
+  success: {
+    label: "Success",
+    color: "hsl(var(--chart-3))",
```
Comment (chronark) [inline apps/dashboard/app/(app)/logs/components/charts/index.tsx:17]: fine for now, but we gotta switch these to the new 12 step color scale
https://engineering.unkey.com/design/colors

File: apps/dashboard/app/(app)/logs/page.tsx [hunk 4]
```diff
@@ -24,25 +24,58 @@ export default async function Page({
       and(eq(table.tenantId, tenantId), isNull(table.deletedAt)),
   });
 
-  if (!workspace?.betaFeatures.logsPage) {
-    return notFound();
-  }
+  // if (!workspace?.betaFeatures.logsPage) {
```
Comment (chronark) [inline apps/dashboard/app/(app)/logs/page.tsx]: put this back please

File: apps/dashboard/lib/trpc/routers/logs/query-timeseries.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      .findFirst({
+        where: (table, { and, eq, isNull }) =>
+          and(eq(table.tenantId, ctx.tenant.id), isNull(table.deletedAt)),
+      })
+      .catch((_err) => {
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          //TODO: change error message later
+          message:
+            "We are unable opt you in to this beta feature. Please contact support using support@unkey.dev",
```
Comment (chronark) [inline apps/dashboard/lib/trpc/routers/logs/query-timeseries.ts:21]: wrong user facing message

**Review by chronark — CHANGES_REQUESTED** [review]

File: apps/dashboard/app/(app)/logs/components/charts/hooks.ts (hunk 1 above)
Comment (ogzhanolguncu) [reply apps/dashboard/app/(app)/logs/components/charts/hooks.ts:15]: We calculate the distance between start and end then decide on which granularity to show

File: apps/dashboard/app/(app)/logs/components/charts/hooks.ts (hunk 2 above)
Comment (ogzhanolguncu) [reply apps/dashboard/app/(app)/logs/components/charts/hooks.ts:30]: I guess this one is a bit redundant, coz trpc diffs them too. We can omit

File: apps/dashboard/app/(app)/logs/components/charts/index.tsx (hunk 3 above)
Comment (ogzhanolguncu) [reply apps/dashboard/app/(app)/logs/components/charts/index.tsx:17]: Will do when we move to design of this page

**Review by chronark — APPROVED** [review]

**Pull request merged**